addons/hr_holidays/models/hr_leave.py

```diff
@@ -359,21 +352,32 @@ def _compute_state(self):
         for leave in self:
             leave.state = 'confirm' if leave.validation_type != 'no_validation' else 'draft'
 
+    @api.depends('holiday_type', 'employee_id', 'department_id', 'mode_company_id')
+    def _compute_resource_calendar_id(self):
+        for leave in self:
+            calendar = False
+            if leave.holiday_type == 'employee':
+                calendar = leave.employee_id.resource_calendar_id
+            elif leave.holiday_type == 'department':
+                calendar = leave.department_id.company_id.resource_calendar_id
+            elif leave.holiday_type == 'company':
+                calendar = leave.mode_company_id.resource_calendar_id
+
+            leave.resource_calendar_id = calendar or self.env.company.resource_calendar_id
```

Comment:
I do see your point, but on the other a leave needs a resource calendar to compute start/end dates and duration. So do you suggest throwing an error here, stating that the employee needs a calendar before they can take leaves?

Reply:
I think it would be best indeed to do that then :)

Reply:
In that case I would make a separate task out of this, so it can be discussed with Geoffrey and potentially others. For the scope of the refactor I think it's best to leave this behaviour as it was and then if we want to improve it, we can have that discussion.